pandas/core/indexes/base.py

```diff
@@ -3888,6 +3888,15 @@ def reindex(
             if self.equals(target):
                 indexer = None
             else:
+                if not self.is_unique:
```

Comment:
don't you need to do this on L3912? e.g. raise if those conditions are tru first. *also* don't we *know* that its unique at this point?

Reply:
> don't you need to do this on L3912? e.g. raise if those conditions are tru first

sure, will move this down

> also don't we know that its unique at this point?

nope